lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -720,7 +823,7 @@ protected function prepare_item_for_database( $request ) {
 		}
 
 		if ( isset( $request['parent'] ) ) {
-			$prepared_comment['comment_parent'] = $request['parent'];
+			$prepared_comment['comment_parent'] = (int) $request['parent'];
```

Comment:
The `get_collection_params()` defines it as an array, so you can use `parent__in` of the `WP_Comment_Query` in `get_items()`. So I guess, the best would be to use `(int)` and make sure its an integer when it comes to writing the database, as I've did.